### ISSUE TITLE
[Proof] Update post-scaffolding for the `Proof` type

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/poktroll#228]
## Objective

- Prepare the codebase for #141 

## Goals

- Prepare the codebase to implement MVP logic for `SubmitProof`
- Put basic validation, plumbing and other business logic in place to make the core business logic easier 
- Do the "busy work" so we can focus on the interesting and important work of `Proof submission`

## Deliverables

- [ ] Similar to how #148 and #151 were implemented for the `Claim` type after #149 was merged, do the same thing for `Proofs`
- [ ] Increase the test coverage for `Claim` along the way if any missing gaps were identified

## Non-goals / Non-deliverables

- Implement the core business logic for Proffs
- Make any business logic for Claims
- Modify any off-chain business logic.

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @Olshansk 
**Co-Owners**: @bryanchriswhite
